sphinx/writers/latex.py

```diff
@@ -521,6 +521,10 @@ def render(self, template_name: str, variables: dict[str, Any]) -> str:
                                  template_name)
             if path.exists(template):
                 return renderer.render(template, variables)
+            elif template_name.endswith('.jinja'):
+                legacy_template = template[:-len('.jinja')] + '_t'
+                if path.exists(legacy_template):
+                    return renderer.render(legacy_template, variables)
```

Comment:
This legacy loading path logic that I switched in isn't something I'm 100% keen on - it's non-obvious enough to feel a bit ugly behaviour-wise.

I think I'll add a warning message when this legacy path is followed.

Reply:
There's also a search precedence question here; do we search all template paths for the `.jinja` file and then all paths for `_t`, or do we (as currently implemented) perform a single pass through the template paths and check for both file variants.

Reply:
Basically I'm worrying about a security-type concern here: it's that someone could place a `_t` (legacy) file in a template path that they control, and that this logic would then load then instead of the expected `.jinja` file.

However: I'm not sure that that particular concern is well-founded, because if an attacker has write access to one of the template paths already, they could equally just write a `.jinja` file in there, and that would achieve the same effect.

However I do think a warning message would still add some value as it could help people to encourage legacy templates to migrate.

Reply:
Given that I'm considering the security concern a non-issue, I think that the file-load-order precedence is less important here, and it is OK to use a logically-simpler single search pass over the template paths.